knowledge_repo/app/routes/posts.py

```diff
@@ -172,3 +177,33 @@ def render_legacy():
 def about():
     """Renders about page. This is the html version of REAMDE.md"""
     return render_template("about.html")
+
+
+@blueprint.route('/ajax/post/download', methods=['GET'])
+@PageView.logged
+def download():
+    "Downloads resources associated with a post."
+
+    path = request.args.get('post')
+    post = current_repo.post(path)
+
+    resource_type = request.args.get('type', 'source')
+    post = current_repo.post(path)
+
+    if resource_type in ('kp', 'zip'):
```

Comment:
It is. I want to keep them separate because in the future I want to link the `.kp` extension to knowledge post for clarity. But since the tooling side of things does not support that yet, it's hidden in the interface :).